app/src/main/java/com/ichi2/apisample/ui/MainActivity.java

```diff
@@ -990,6 +1013,37 @@ public void onClick(DialogInterface dialog, int id) {
                         }
                     })
                     .show();
+        } catch (final MusInterval.DefaultModelOutdatedException e) {
+            final String modelName = e.getModelName();
+            new AlertDialog.Builder(this)
+                    .setMessage(String.format(
+                            getResources().getString(R.string.update_model),
+                            modelName))
+                    .setPositiveButton(R.string.update, new DialogInterface.OnClickListener() {
+                        public void onClick(DialogInterface dialog, int id) {
+                            Long updatedModelId = mAnkiDroid.updateCustomModel(
+                                    mAnkiDroid.findModelIdByName(modelName),
```

Comment:
Since you already had the "model_id" value right before exception throwing, I guess it can be also transferred in exception

Reply:
Good point. Done.